src/main/core/work/event.c

```diff
@@ -50,28 +49,14 @@ void event_execute(Event* event, Host* host) {
 
     utility_assert(event_getHostID(event) == host_getID(host));
 
-    /* check if we are allowed to execute or have to wait for cpu delays */
-    CPU* cpu = host_getCPU(host);
-    cpu_updateTime(cpu, event->time);
-
-    if(cpu_isBlocked(cpu)) {
-        SimulationTime cpuDelay = cpu_getDelay(cpu);
-        trace("event blocked on CPU, rescheduled for %"G_GUINT64_FORMAT" nanoseconds from now", cpuDelay);
-
-        /* track the event delay time */
-        Tracker* tracker = host_getTracker(host);
-        if (tracker != NULL) {
-            tracker_addVirtualProcessingDelay(tracker, cpuDelay);
-        }
+    host_continueExecutionTimer(host);
+    taskref_execute(event->task, host);
+    host_stopExecutionTimer(host);
+}
 
-        /* this event is delayed due to cpu, so reschedule it to ourselves */
-        worker_scheduleTaskWithDelay(event->task, host, cpuDelay);
-    } else {
-        /* cpu is not blocked, its ok to execute the event */
-        host_continueExecutionTimer(host);
-        taskref_execute(event->task, host);
-        host_stopExecutionTimer(host);
-    }
+TaskRef* event_borrowTask(Event* event) {
```

Comment:
Agreed, but since this was removed in the next commit where the C version was replaced with a rust version and `event_borrowTask` became `event_intoTask()`, I'm going to leave it as-is.